### PR TITLE
8303511: C2: assert(get_ctrl(n) == cle_out) during unrolling

### PR DESCRIPTION
**Pull request by rwestrel**
In the same round of loop optimizations:

- `PhaseIdealLoop::remix_address_expressions()` creates a new `AddP`
  out of loop. It sets it control to
  `n_loop->_head->in(LoopNode::EntryControl)` which, because the loop is
  strip mined, is an `OuterStripMinedLoop`.
  
- The `LoadI` for that `AddP` is found to only have uses outside the
  loop and is cloned out of the loop. It's referenced by the outer
  loop's safepoint.
  
- The loop is unrolled. Unrolling follows the safepoint's inputs and
  find the new `AddP` with control set to the `OuterStripMinedLoop`
  and the assert fires.
  
No control should be set to an `OuterStripMinedLoop`. The fix is
straightforward and sets the control to the `OuterStripMinedLoop`
entry control.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8303511](https://bugs.openjdk.org/browse/JDK-8303511): C2: assert(get_ctrl(n) == cle_out) during unrolling

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to [c2787438](https://git.openjdk.org/jdk/pull/12824/files/c2787438192b78dfad1c1c7a2427fd31c5e2c688)
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12824/head:pull/12824` \
`$ git checkout pull/12824`

Update a local copy of the PR: \
`$ git checkout pull/12824` \
`$ git pull https://git.openjdk.org/jdk pull/12824/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12824`

View PR using the GUI difftool: \
`$ git pr show -t 12824`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12824.diff">https://git.openjdk.org/jdk/pull/12824.diff</a>

</details>

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Yes, this looks good.

File: test/hotspot/jtreg/compiler/loopstripmining/TestAddPAtOuterLoopHead.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        int m;
+        for (m = 0; m < 2; m++) {
+            for (int i = 0; i < 2; i++) {
+                for (int j = 0; j < 2; j++) {
+                }
+            }
+        }
+
+
+        int v  = 0;
```
Comment (turbanoff) [inline test/hotspot/jtreg/compiler/loopstripmining/TestAddPAtOuterLoopHead.java:65]:
```suggestion
        int v = 0;
```

Comment (rwestrel) [reply test/hotspot/jtreg/compiler/loopstripmining/TestAddPAtOuterLoopHead.java]: thanks for catching that.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by rwestrel** [comment]
@vnkozlov @TobiHartmann thanks for the reviews!

**Pull request closed without merging**